portal/server/utils/csv-utils.js

```diff
@@ -41,6 +41,32 @@ const excelColumnToColumnIndex = (column) => {
   return result - 1;
 };
 
+/**
+ * Checks for floating-point rounding errors and rounds to two decimal places if within error tolerance.
+ * @param {number} number - The number to check and potentially round.
+ * @returns {number} - The rounded number to two decimal places if within tolerance, otherwise the original number.
+ * @throws {TypeError} - If the input is not a number.
+ */
+const handleFloatingPointRoundingErrors = (number) => {
+  if (typeof number !== 'number') {
+    throw new TypeError('Input must be a number');
+  }
+
+  const TOLERANCE = 1e-6;
+  const DECIMAL_PLACES = 2;
+
+  const roundedNumber = Number(number.toFixed(DECIMAL_PLACES));
```

Comment:
Out of interest -- have you considered calling `Number(number.toPrecision(2))`?

Not looked into it much however...

 https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toPrecision

Reply:
Ah my bad, thanks :)

Reply:
Good suggestion.